traffic_ops/testing/api/v3/servicecategories_test.go

```diff
@@ -0,0 +1,137 @@
+package v3
+
+/*
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"testing"
+	"time"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	toclient "github.com/apache/trafficcontrol/traffic_ops/client"
+)
+
+func TestServiceCategories(t *testing.T) {
+	WithObjs(t, []TCObj{ServiceCategories, Tenants, Users}, func() {
+		UpdateTestServiceCategories(t)
+		GetTestServiceCategories(t)
+		ServiceCategoryTenancyTest(t)
+	})
+}
+
+func CreateTestServiceCategories(t *testing.T) {
+	// loop through service categories, assign FKs and create
+	for _, sc := range testData.ServiceCategories {
+		resp, _, err := TOSession.CreateServiceCategory(sc)
+		if err != nil {
+			t.Errorf("could not CREATE service category: %v", err)
+		}
+		t.Log("Response: ", resp.Alerts)
+	}
+}
+
+func GetTestServiceCategories(t *testing.T) {
+	for _, sc := range testData.ServiceCategories {
+		resp, _, err := TOSession.GetServiceCategoryByName(sc.Name)
+		if err != nil {
+			t.Errorf("cannot GET Service Category by name: %v - %v", err, resp)
+		}
+	}
+}
+
+func UpdateTestServiceCategories(t *testing.T) {
+	firstServiceCategory := testData.ServiceCategories[0]
+	// Retrieve the Service Category by service category so we can get the id for the Update
+	resp, _, err := TOSession.GetServiceCategoryByName(firstServiceCategory.Name)
+	if err != nil {
+		t.Errorf("cannot GET Service Category by service category: %v - %v", firstServiceCategory.Name, err)
+	}
+	if len(resp) > 0 {
+		remoteServiceCategory := resp[0]
+		expectedServiceCategory := "service-category-test"
```

Comment:
This is a bit of a nitpick, but this test could give false positives if the category's name was already "service-category-test". Instead it's generally better to concatenate something to the original value, so that you know for sure it's different no matter what it was initially.